### DISCUSSION TITLE
Add functionality to save attack models.

### DISCUSSION THREAD

**Discussion by ROYALBEFF** [Trusted-AI/adversarial-robustness-toolbox#1542] [Ideas]
I'm currently running multiple experiments and want to store the trained attack models for later use, similar to `torch.save()`.
I couldn't find anything in the documentation, so I assume it's not yet possible to store attack models.

This is a crucial feature for any ML library!

**Comment by beat-buesser** [comment]
Hi @ROYALBEFF

Thank you very much for using ART! I think that could be a useful feature.

Could you please describe your use case in more detail?

Would be interested to implement a prototype?

**Reply by ROYALBEFF** [reply]
Hi @beat-buesser.

The use-case is rather simple. I run several experiments where I train a model and a corresponding attack model, which in my case is a  membership inference attack. Then I attack the original model using different attack datasets. At this point, I want to store the attack model for two reasons:

1) I want my experiments to be reproducible, such that others can easily comprehend the results. This is very important for scientific research.
2) I want to be able to attack the original model again with another attack dataset without training the attack model from scratch again.

I can try my best implementing a prototype, but at the moment I have only little time left for it.
